pageserver/src/tenant/secondary.rs

```diff
@@ -254,6 +252,22 @@ impl SecondaryTenant {
         .await
         .expect("secondary eviction should not have panicked");
     }
+
+    /// Exhaustive check that incrementally updated metrics match the actual state.
+    #[cfg(feature = "testing")]
+    fn validate_metrics(&self) {
+        if cfg!(feature = "testing") {
```

Comment:
nit: why check for the feature again here?

Reply:
oops, that's a copy paste error.  will fix.